### PR TITLE
add api_keys and other refactors for startup pkg

### PR DESCRIPTION
**Pull request by RazcoDev**

### CODE REVIEW

**Review by danyi1212 — APPROVED** [review]
Looks good, left few cosmetic comments

**Comment by danyi1212** [comment]
Are there any integration tests for that logic? Couldn't find any unit tests here

File: horizon/facts/client.py [hunk 1]
```diff
@@ -21,7 +22,7 @@ def client(self) -> AsyncClient:
         if self._client is None:
             self._client = AsyncClient(
                 base_url=sidecar_config.CONTROL_PLANE,
-                headers={"Authorization": f"Bearer {sidecar_config.API_KEY}"},
+                headers={"Authorization": f"Bearer {get_env_api_key()}"},
```
[the remark this answers is not in the document]
Comment (RazcoDev) [reply horizon/facts/client.py:25]: done

File: horizon/opal_relay_api.py [hunk 2]
```diff
@@ -96,7 +97,7 @@ def _apply_context(self, context: dict[str, str]):
     def api_session(self) -> ClientSession:
         if self._api_session is None:
             self._api_session = ClientSession(
-                headers={"Authorization": f"Bearer {sidecar_config.API_KEY}"}
+                headers={"Authorization": f"Bearer {get_env_api_key()}"}
```
[the remark this answers is not in the document]
Comment (RazcoDev) [reply horizon/opal_relay_api.py:100]: done

File: horizon/startup/api_keys.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        "reraise": True,
+    }
+
+    def __init__(
+        self,
+        backend_url: str = sidecar_config.CONTROL_PLANE,
+        retry_config=None,
+    ):
+        self._backend_url = backend_url
+        self._retry_config = retry_config or self.DEFAULT_RETRY_CONFIG
```
[the remark this answers is not in the document]
Comment (RazcoDev) [reply horizon/startup/api_keys.py:27]: Leaving it like this for consistency with the remote_config class

**Pull request merged**